libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1738,7 +1751,7 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 					argType->mobileType() &&
 					argType->mobileType()->interfaceType(false) &&
 					argType->mobileType()->interfaceType(false)->encodingType() &&
-					!(dynamic_cast<StructType const*>(argType->mobileType()->interfaceType(false)->encodingType().get()))
+					(abiEncodeV2 || !(dynamic_cast<StructType const*>(argType->mobileType()->interfaceType(false)->encodingType().get())))
```

Comment:
This should probably have an additional check: Structs are fine if abiv2 is used, but only if compact encoding is not used.

Reply:
Added the check.